### PR TITLE
Fix wrong getAll keyValue method

### LINKED ISSUE #8520: Onboarding 🐛 : Infinite loop if I want to sync Emails and Calendar
## Bug Description

During the Onboarding, I encountered an issue trying to sync emails and calendar.
Screenshot 2024-11-15 at 17 04 54

**Steps to reproduce** 
- ask someone to create an account for you 
- click on the email you received
- signup
- click "subject and metadata" and sync your calendar and emails
- click authorize on the next screens
- you end up again on the same screen, The loop starts again... so click "subject and metadata" and sync your calendar and emails...

## Expected behavior

You should end up in the hompage after the onboarding

## Technical inputs

Tested in production, cloud env from the twenty core team. Email Invitation to join the team was done by @FelixMalfait 
Browser : Chrome latest version

### PR DESCRIPTION
**Pull request by martmull**
Fixes https://github.com/twentyhq/twenty/issues/8520#issuecomment-2485913500

- Fix the getAll key-value pairs method, as it returns incorrect results. The keyVar values were being affected by keyValues from other workspaces.
- Fix missing Sync Email step for existing users joining existing workspace
  
Before, this method returned keyValues that verifies:
- workspaceId = user.defaultWorkspaceId
- userId = user.id

This method now returns:
- userId = null && workspaceId = user.defaultWorkspaceId
- userId = user.id && workspaceId = null
- userId = user.id && workspaceId = user.defaultWorkspaceId

## Result

https://github.com/user-attachments/assets/b7563db6-a6a8-4e09-a0c6-c372d7e2b712

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts [hunk 1]
```diff
@@ -75,32 +76,32 @@ export class UserVarsService<
     ]).get(key) as KeyValueTypesMap[K];
   }
 
-  public async getAll({
-    userId,
-    workspaceId,
-  }: {
-    userId?: string;
-    workspaceId?: string;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts:83]: keep this in the signature

File: packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts [hunk 2]
```diff
@@ -75,32 +76,32 @@ export class UserVarsService<
...
     });
 
-    let userVarsUserLevel: any[] = [];
-
-    if (userId) {
-      userVarsUserLevel = await this.keyValuePairService.get({
-        type: KeyValuePairType.USER_VAR,
-        userId,
-      });
+    if (user.defaultWorkspaceId) {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts:88]: revert and only add workspaceId: null to compute userVarsUserLevel

File: packages/twenty-server/src/engine/core-modules/user/user.resolver.ts [hunk 3]
```diff
@@ -75,20 +79,14 @@ export class UserResolver {
   }
 
   @ResolveField(() => GraphQLJSONObject)
-  async userVars(
-    @Parent() user: User,
-    @AuthWorkspace() workspace: Workspace,
-  ): Promise<Record<string, any>> {
-    const userVars = await this.userVarService.getAll({
-      userId: user.id,
-      workspaceId: workspace?.id ?? user.defaultWorkspaceId,
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/user/user.resolver.ts:84]: remove ?? user.defaultWorkspaceId indeed

**Review by charlesBochet — APPROVED** [review]

File: packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts (hunk 2 above)
Comment (martmull) [reply packages/twenty-server/src/engine/core-modules/user/user-vars/services/user-vars.service.ts:88]: we will miss userId non null, and workspaceId non null variables

**Review by guillim — APPROVED** [review]
Could not properly reproduce locally, but code SGTM

**Pull request merged**